### PR TITLE
Add memberships table in user groups page

### PR DESCRIPTION
**Pull request by AdityaKhatri**
- Addresses https://zube.io/deep/deep/c/3724
- Depends on: https://github.com/the-deep/server/pull/884

## Changes

* Add paginated memberships data

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] build works
- [x] eslint issues
- [x] typescript issues
- [x] `console.log` meant for debugging
- [x] typos
- [x] unwanted comments
- [x] conflict markers

## This PR contains valid:

- [x] permission checks
- [x] translations

### CODE REVIEW

File: src/newViews/UserGroup/Memberships/index.tsx [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+                keySelector={membershipKeySelector}
+                data={memberships?.results}
+            />
+            {showAddUserModal && userGroup && (
+                <AddUserModal
+                    onModalClose={setUserModalHidden}
+                    group={userGroup}
+                    onUserAddSuccess={usersGetTrigger}
+                    userToEdit={userToEdit}
+                    memberOptions={memberOptions}
```
Comment (tnagorra) [inline src/newViews/UserGroup/Memberships/index.tsx]: Why pass memberOptions?

Comment (AdityaKhatri) [reply src/newViews/UserGroup/Memberships/index.tsx]: The same form is used to get user details. By default the list needs to have the current user so we are sending the list.
We could send only the selected user, but its the same.

Comment (tnagorra) [reply src/newViews/UserGroup/Memberships/index.tsx]: Is that to preselect current user on the form?

Comment (tnagorra) [reply src/newViews/UserGroup/Memberships/index.tsx]: The prop name should be renamed.
Plus I noticed that the usage in AddUserModal is incorrect.

**Review by tnagorra — CHANGES_REQUESTED** [review]

**Review by tnagorra — APPROVED** [review]

**Review by tnagorra — APPROVED** [review]

**Review by subinasr — APPROVED** [review]

**Pull request merged**